src/prefect_server/api/flows.py

```diff
@@ -355,9 +356,26 @@ async def archive_flow(flow_id: str) -> bool:
 
     # delete scheduled flow runs
     await models.FlowRun.where(
-        {"flow_id": {"_eq": flow_id}, "state": {"_eq": "Scheduled"}}
+        {
+            "flow_id": {"_eq": flow_id},
+            "state": {"_eq": "Scheduled"},
+            "auto_scheduled": {"_eq": True},
+        }
     ).delete()
 
+    # cancel adhoc scheduled flow runs
+    fr_ids = await models.FlowRun.where(
+        {
+            "flow_id": {"_eq": flow_id},
+            "state": {"_eq": "Scheduled"},
+            "auto_scheduled": {"_eq": False},
+        }
+    ).get({"id"})
+    msg = f"Flow {flow_id} was archived."
+    tasks = []
+    for flow_run in fr_ids:
+        tasks.append(api.states.cancel_flow_run(flow_run.id, state_message=msg))
+    await asyncio.gather(*tasks)
```

Comment:
I have a preference for a list compr since it's not reused but 🤷 
```suggestion
    await asyncio.gather(
        *[
            api.states.cancel_flow_run(flow_run.id, state_message=msg)
            for flow_run in fr_ids
        ]
    )
```

Reply:
works for me - will update in a different commit because I forgot to import `asyncio` 🤦